include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -538,6 +561,141 @@ make_transform_iterator(_Iter __it, _UnaryFunc __unary_func)
     return transform_iterator<_Iter, _UnaryFunc>(__it, __unary_func);
 }
 
+template <typename _Iter, typename _UnaryFunc>
+class transform_output_iterator
+{
+  private:
+    _Iter __my_it_;
+    const _UnaryFunc __my_unary_func_;
+
+  public:
+    typedef typename ::std::iterator_traits<_Iter>::difference_type difference_type;
+    typedef typename ::std::iterator_traits<_Iter>::value_type value_type;
+    typedef __internal::transform_output_ref_wrapper<value_type, decltype(__my_unary_func_)> reference;
+    typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
+    typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
+
+    transform_output_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
+        : __my_it_(__it), __my_unary_func_(__unary_func)
+    {
+    }
+    transform_output_iterator(const transform_output_iterator& __input) = default;
+    transform_output_iterator&
+    operator=(const transform_output_iterator& __input)
+    {
+        __my_it_ = __input.__my_it_;
```

Comment:
From my point of view, it's seems like an error that you don't copy __my_unary_func_ here.

Reply:
> From my point of view, it's seems like an error that you don't copy __my_unary_func_ here.

Only if a functor keeps some state.. I case of stateless  - no error.

Reply:
If it's stateless - may be not required to have it as class member?

Reply:
There's no restriction placed on the type by other libraries that implement a similar class. A use case of a stateful functor would be a `multiply_by` struct with a member value that was the factor by which all elements read would be multiplied.  I think the `operator=` should be updated with the assignment as Sergey suggests.